crypto/ube/ube_test.cc

```diff
@@ -53,92 +33,39 @@ TEST_F(ubeTest, BasicTests) {
 
   uint64_t generation_number = 0;
   uint64_t cached_generation_number = 0;
+  uint32_t mocked_generation_number = 0;
 
-  // First test incrementing the fork generation number. Pick a starting point
-  // and get initial UBE generation number
-  set_fork_generation_number_FOR_TESTING(5);
-  ASSERT_TRUE(CRYPTO_get_ube_generation_number(&generation_number));
-
-  // Should be stable again.
-  cached_generation_number = generation_number;
-  generation_number = 0;
-  ASSERT_TRUE(CRYPTO_get_ube_generation_number(&generation_number));
-  ASSERT_EQ(generation_number, cached_generation_number);
-
-  // Mock a snapsafe event.
-  set_snapsafe_generation_number_FOR_TESTING(6);
-
-  // Generation number should have incremented once.
-  cached_generation_number = generation_number;
-  generation_number = 0;
-  ASSERT_TRUE(CRYPTO_get_ube_generation_number(&generation_number));
-  ASSERT_EQ(generation_number, cached_generation_number + 1);
-
-  // Should be stable again.
-  cached_generation_number = generation_number;
-  generation_number = 0;
-  ASSERT_TRUE(CRYPTO_get_ube_generation_number(&generation_number));
-  ASSERT_EQ(generation_number, cached_generation_number);
-
-  // Mock another snapsafe event with higher increment.
-  set_snapsafe_generation_number_FOR_TESTING(8);
-
-  // Generation number should have incremented once.
-  cached_generation_number = generation_number;
-  generation_number = 0;
-  ASSERT_TRUE(CRYPTO_get_ube_generation_number(&generation_number));
-  ASSERT_EQ(generation_number, cached_generation_number + 1);
-
-  // Should be stable again.
-  cached_generation_number = generation_number;
-  generation_number = 0;
-  ASSERT_TRUE(CRYPTO_get_ube_generation_number(&generation_number));
-  ASSERT_EQ(generation_number, cached_generation_number);
-
-  // Mock another snapsafe event but with a strictly smaller value.
-  set_snapsafe_generation_number_FOR_TESTING(1);
-
-  // Generation number should have incremented once.
-  cached_generation_number = generation_number;
-  generation_number = 0;
-  ASSERT_TRUE(CRYPTO_get_ube_generation_number(&generation_number));
-  ASSERT_EQ(generation_number, cached_generation_number + 1);
-
-  // Should be stable again.
-  cached_generation_number = generation_number;
-  generation_number = 0;
-  ASSERT_TRUE(CRYPTO_get_ube_generation_number(&generation_number));
-  ASSERT_EQ(generation_number, cached_generation_number);
-}
-
-TEST_F(ubeTest, MockedMethodTestsFork) {
-
-  allowMockedUbe();
-
-  uint64_t generation_number = 0;
-  uint64_t cached_generation_number = 0;
+  uint8_t initial_mocked_generation_number[4] = {0};
+  ASSERT_TRUE(RAND_bytes(initial_mocked_generation_number, 4));
+  mocked_generation_number =
+        ((uint32_t)initial_mocked_generation_number[0] << 24) |
+        ((uint32_t)initial_mocked_generation_number[1] << 16) |
+        ((uint32_t)initial_mocked_generation_number[2] << 8)  |
+        ((uint32_t)initial_mocked_generation_number[3]);
```

Comment:
I haven't thought about generating a number instead of just bytes for a while. Just curious is this equivalent to something like:
```
  RAND_bytes(&generation_number, sizeof(uint64_t));
```

Reply:
Well I guess one could maybe do
```
RAND_bytes((uint8_t*) &mocked_generation_number, 4);
```
or something